### PR TITLE
Create YumRepository with repodataDepth field

### PR DESCRIPTION
**Pull request by gmarraffa**
Before this commit the depth kwarg was ignored in the actual creation of the repository
With this fix the value is correctly bundled in the configuration

### CODE REVIEW

**Comment by gmarraffa** [comment]
I wrote a test to check the correct value in the configuration but I'm not really capable of writing the integration test to check if the value is really inserted in the system. Also I don't think that Nexus API show that particular field in the reponse.

By the way, I manually checked that the value is correctly inserted in the configuration of the repository via the Nexus GUI.
Further step includes to check that the depth arg, at construction time, is between 0 and 5 (range for yum repos), since if the depth is not in the range it will be ignored while creating the repository.

**Comment by gmarraffa** [comment]
@thiagofigueiro any update on this one?

**Comment by thiagofigueiro** [comment]
hi, @gmarraffa - looking at it now! :-)

Thanks for the patience. My github notifications were disabled and I didn't realise this was sitting here for so long. My apologies.

**Comment by gmarraffa** [comment]
No worries! Thank you for taking care of this repo!

File: src/nexuscli/api/repository/model.py [hunk 1]
```diff
@@ -418,6 +418,20 @@ def __init__(self, name, depth=1, **kwargs):
...
+    @property
+    def configuration(self):
+        """
+        As per :py:obj:`Repository.configuration` but specific to this
+        repository recipe and type.
+
+        :rtype: str
+        """
+        repo_config = super().configuration
+        repo_config["attributes"]["yum"] = {
```
Comment (thiagofigueiro) [inline src/nexuscli/api/repository/model.py]: nitpick: the rest of the codebase primarily uses single quotes.
```suggestion
        repo_config['attributes']['yum'] = {
```

File: src/nexuscli/api/repository/model.py [hunk 2]
```diff
@@ -418,6 +418,20 @@ def __init__(self, name, depth=1, **kwargs):
...
+            "repodataDepth": self.depth
```
Comment (thiagofigueiro) [inline src/nexuscli/api/repository/model.py]:
```suggestion
            'repodataDepth': self.depth
```

**Pull request merged**